modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/web/cache/BaseAsahWebCacheItem.java

```diff
@@ -0,0 +1,107 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.search.internal.web.cache;
+
+import com.liferay.analytics.settings.configuration.AnalyticsConfiguration;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.json.JSONFactoryUtil;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.util.Http;
+import com.liferay.portal.kernel.util.HttpUtil;
+import com.liferay.portal.kernel.util.Validator;
+import com.liferay.portal.kernel.webcache.WebCacheItem;
+
+import java.net.HttpURLConnection;
+
+/**
+ * @author Gustavo Lima
+ */
+public abstract class BaseAsahWebCacheItem implements WebCacheItem {
```

Comment:
I'm not sure if having this base class is necessary or makes things cleaner assuming the convert() becomes abstract or empty. Only the validate() remains.
Then, it could  be possible to make this cache class to serve all AC requests by building the API url before entering the class. Maybe a matter of taste.

Reply:
I add one commit that basically refactor the code to use just on AsahWebCacheItem to both